compiler/rustc_hir/src/hir.rs

```diff
@@ -571,6 +571,10 @@ impl<'hir> Generics<'hir> {
         self.params.iter().find(|&param| name == param.name.ident().name)
     }
 
+    pub fn get_with_def_id(&self, id: DefId) -> Option<&GenericParam<'hir>> {
```

Comment:
This is definitely not correct if `DefId` comes from a different crate. Please make this take a `LocalDefId`, or ideally, do the comparison with `HirId` instead.

Reply:
I wad not able to find a way to convert a DefId to a HirId, so I used a LocalDefId instead. The place where I use it already has expect_local() on the binding scope, so I assume that the lifetime must also be local.